### ISSUE TITLE
Incorrect available versions with empty versions range

### ISSUE THREAD

**Issue by konstin** [astral-sh/uv#3307]
I requested `ruff>=0.4.3,<0.5`, while 0.4.2 is currently the latest version. This shows:
```
  x No solution found when resolving dependencies:
  `-> Because only the following versions of ruff are available:
          ruff<0.4.3
          ruff>=0.5
      and gha-api-test depends on ruff>=0.4.3,<0.5, we can conclude that the requirements are unsatisfiable.
```
That is incorrect, `ruff>=0.5` is not available.

**Comment by zanieb** [comment]
That's... weird. Thanks!

**Comment by zanieb** [comment]
Still wrong

```
❯ uv pip install 'ruff>=0.5.8,<0.6'
  × No solution found when resolving dependencies:
  ╰─▶ Because only the following versions of ruff are available:
          ruff<0.5.8
          ruff>0.6
      and you require ruff>=0.5.8,<0.6, we can conclude that your requirements are unsatisfiable.
```

**Comment by zanieb** [comment]
Actually not still wrong!

```
❯ uv pip install 'ruff>=0.6.10,<0.7'
  × No solution found when resolving dependencies:
  ╰─▶ Because only ruff<=0.6.9 is available and you require ruff>=0.6.10,<0.7, we can conclude that your requirements are
      unsatisfiable.
```

**Comment by zanieb** [comment]
Solved in #6118
